### PR TITLE
Make it so licenses are sourced from a shared file

### PR DESCRIPTION
**Pull request by rfbgo**
Previously, licenses from configs/licenses.yaml were added directly into the execute_experiment script. This meant you repeated the licenses per file, and could not easily update them without re-setting up a job

This change adds the folder `$workspace/shared/licneses/$app` and generates the file `licenses.inc` where the env sourcing commands are stored. Each experiment can then source this file

### CODE REVIEW

**Review by douglasjacobsen — APPROVED** [review]
This LGTM. Just a few small comments, but they aren't necessary and are more things we should think about than things that need changing right here.

File: lib/ramble/ramble/workspace/workspace.py [hunk 1]
```diff
@@ -660,6 +673,8 @@ def write(self):
         fs.mkdirp(self.experiment_dir)
         fs.mkdirp(self.input_dir)
         fs.mkdirp(self.software_dir)
+        fs.mkdirp(self.shared_dir)
```
[the remark this answers is not in the document]
Comment (rfbgo) [reply lib/ramble/ramble/workspace/workspace.py:676]: Yea I like that suggestion

File: lib/ramble/ramble/application_types/executable.py [hunk 2]
```diff
@@ -21,12 +21,10 @@ class ExecutableApplication(ApplicationBase):
 
     def __init__(self, file_path):
         super().__init__(file_path)
-        self._setup_phases = [
+        self._setup_phases += [
```
[the remark this answers is not in the document]
Comment (rfbgo) [reply lib/ramble/ramble/application_types/executable.py:24]: Discussed offline, we will leave this as is and later refactor phases to express order dependence

File: lib/ramble/ramble/application_types/spack.py [hunk 3]
```diff
@@ -46,7 +46,7 @@ class SpackApplication(ApplicationBase):
 
     def __init__(self, file_path):
         super().__init__(file_path)
-        self._setup_phases = [
+        self._setup_phases += [
```
[the remark this answers is not in the document]
Comment (rfbgo) [reply lib/ramble/ramble/application_types/spack.py:49]: Discussed offline, we will leave this as is and later refactor phases to express order dependence

**Pull request merged**